plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/KuduModule.java

```diff
@@ -107,4 +155,15 @@ KuduClientSession createKuduClientSession(KuduClientConfig config)
         }
         return new KuduClientSession(client, strategy);
     }
+
+    private static KuduClient getKerberizedKuduClient(KuduClient.KuduClientBuilder builder, KuduKerberosConfig kuduKerberosConfig)
+    {
+        kuduKerberosConfig.getKuduPrincipalPrimary().ifPresent(builder::saslProtocolName);
+        setJavaSecurityKrb5Conf(kuduKerberosConfig.getConfig().getAbsolutePath());
+        KerberosAuthentication kerberosAuthentication = new KerberosAuthentication(kuduKerberosConfig.getClientPrincipal(), kuduKerberosConfig.getClientKeytab());
+        CachingKerberosAuthentication cachingKerberosAuthentication = new CachingKerberosAuthentication(kerberosAuthentication);
```

Comment:
Right now ticket expiration will not work with this approach of a singleton `KuduClientSession`, looking into options. If people have ideas let me know.